pkg/apis/autoscaling/types.go

```diff
@@ -202,6 +202,62 @@ type HorizontalPodAutoscalerStatus struct {
 
 	// CurrentMetrics is the last read state of the metrics used by this autoscaler.
 	CurrentMetrics []MetricStatus
+
+	// conditions is the set of conditions required for this autoscaler to scale its target,
+	// and indicates whether or not those conditions are met.
+	Conditions []HorizontalPodAutoscalerCondition
```

Comment:
(optional) Have you consider using here map: ConditionType ->  HorizontalPodAutoscalerCondition